packages/venia-ui/lib/components/MyAccount/ResetPassword/resetPassword.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { Form } from 'informed';
+
+import { useResetPassword } from '@magento/peregrine/lib/talons/MyAccount/useResetPassword';
+import { mergeClasses } from '@magento/venia-ui/lib/classify';
+
+import { Title } from '../../Head';
+import Field from '../../Field';
+import TextInput from '../../TextInput';
+import Button from '../../Button';
+import FormErrors from '../../FormError';
+import { isRequired } from '../../../util/formValidators';
+
+import resetPasswordOperations from './resetPassword.gql';
+
+import defaultClasses from './resetPassword.css';
+
+const PAGE_TITLE = `Reset Password`;
+
+const ResetPassword = props => {
+    const { classes: propClasses } = props;
+    const classes = mergeClasses(defaultClasses, propClasses);
+    const talonProps = useResetPassword({
+        ...resetPasswordOperations
+    });
+    const {
+        hasCompleted,
+        loading,
+        email,
+        token,
+        formErrors,
+        handleSubmit
+    } = talonProps;
+
+    const tokenMissing = (
+        <div className={classes.invalidToken}>
+            Token missing. Error message TODO.
```

Comment:
@schensley to provide a proper error message to show when the link is not as expected.